api/api-server-site-urls-config-v1/src/main/java/com/thoughtworks/go/apiv1/serversiteurlsconfig/ServerSiteUrlsConfigControllerV1.java

```diff
@@ -77,8 +82,16 @@ public void setupRoutes() {
 
     String createOrUpdate(Request request, Response response) throws IOException {
         SiteUrls siteUrls = buildEntityFromRequestBody(request);
-        serverConfigService.createOrUpdateServerSiteUrls(siteUrls);
-        return writerForTopLevelObject(request, response, writer -> ServerSiteUrlsConfigRepresenter.toJSON(writer, siteUrls));
+
+        try {
+            serverConfigService.createOrUpdateServerSiteUrls(siteUrls);
+        } catch (GoConfigInvalidException e) {
+            response.status(HttpStatus.UNPROCESSABLE_ENTITY.value());
+            String errorMessage = entityConfigValidationFailed(siteUrls.getClass().getAnnotation(ConfigTag.class).value(), e.getAllErrorMessages());
+            return MessageJson.create(errorMessage, jsonWriter(siteUrls));
+        }
+
+        return index(request, response);
```

Comment:
Any specific reason for using `index()` over `writerForTopLevelObject`?

Reply:
Not really.
As we've same implementation in place under `index` method, thought of using the same.

Reply:
Ok sound logical. But, IMO, we should keep the 2 calls different as in this way, the `create` response is dependent upon the implementation of `index`. If the latter changes, the former might need a change too. WDYT?

Reply:
Agree.
Will update it.